examples/acmetool/hooks/01triggerRancherGen.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+#
+# Executed by acmetool when a certificate is updated
+# Ensures that /etc/nginx/certs/ is present, then
+# overwrite-copies all live SSL certs to that directory.
+# Triggers a single-run of rancher-gen to reload Nginx containers.
+
+case "$1" in
+ "live-updated" )
+    echo "[HOOK]: Processing live-updated hook"
+    mkdir -p /etc/nginx/certs
+    cp -RLf /var/lib/acme/live/* /etc/nginx/certs/
+    rancher-gen --config ${RGEN_CONFIG_ONETIME}
+    ;;
```

Comment:
First, this file is actually deprecated and should now be removed! This hook was introduced to get Rancher-Gen to re-run after certificates were loaded, but it created an extra loop iteration that was not desired.

Old loop:

1. Stack created
1. Rancher-Gen.cfg tpl 1 - generate basic nginx.config (allows nginx containers to come online properly).
1. Rancher-Gen.cfg tpl 2 - Create SSL certs via `notify-cmd="acmetool want {{rgon.domain}}`
1. 01triggerRancherGen.sh - copy certs to destination
1. Rancher-Gen-onetime.cfg - generate new nginx.config with SSL certs
1. Repeat from step 2

So each time the main Rancher-Gen.cfg runs (which is frequently, not _just_ when metadata updates) it would trigger acmetool and reload the `nginx` containers twice.

A better approach is to generate the SSL certs first, then reload `nginx` containers once. However this creates a race condition in that the `nginx` containers can't start immediately as they have to wait for `acmetool` to do an initial cert run. The solution is to run a basic Rancher-Gen pass to create `nginx.conf` quickly, then allow the Rancher-Gen SSL run.

New loop:

1. Stack created
1. Rancher-Gen-firstrun.cfg - generate basic nginx.config (allows nginx containers to come online properly). Only runs once on stack creation.
1. Rancher-Gen.cfg tpl 1 - Create SSL certs
1. Rancher-Gen.cfg tpl 2 - generate new nginx.config with SSL certs
1. Repeat from step 3

Finally, Nginx can't read certificates from symlinked directories - the certs need to be `cp -L` (dereferenced) before use.